services/device-registry/src/main/java/org/eclipse/hono/deviceregistry/FileBasedTenantService.java

```diff
@@ -565,4 +573,15 @@ private String generateTenantId() {
     private boolean checkResourceVersion(final Optional<String> expectedVersion, final String actualValue) {
         return actualValue.equals(expectedVersion.orElse(actualValue));
     }
+
+    private boolean isValidTenantPayload(final JsonObject jsonObject, final Span span) {
+        try {
+            jsonObject.mapTo(Tenant.class);
+            return true;
+        } catch (final IllegalArgumentException e) {
+            log.debug("Error parsing payload of tenant request", e);
+            TracingHelper.logError(span, e);
+            return false;
+        }
+    }
```

Comment:
The request payload is already being checked in https://github.com/eclipse/hono/blob/0999336ef6aaad9c629878046619cc8a78ee376d/service-base/src/main/java/org/eclipse/hono/service/management/tenant/EventBusTenantManagementAdapter.java#L204

I guess you should add the additional check there ...

Reply:
@sophokles73 Thanks for pointing out and I will update here. But it seems to me that if the Tenant API attributes get changed then above validations should be adapted manually. 

IMHO `Tenant.class` acts as a blueprint and the validation can be automated by trying to map the payload with this class as in this PR. For example `hasValidAdapterSpec()` method is not needed if the payload is already validated with `Tenant.class`. Any validations that are not specified in the `Tenant.class` can be added additionally above. WDYT?

Reply:
absolutely :+1:

Reply:
@sophokles73 I have pushed the changes. Would you mind having a look?